app/controllers/users_controller.rb

```diff
@@ -28,7 +30,7 @@ def create
   # and find that user information like username,name,role,email
   def show
     @user = User.shod(params[:id])
-    @student = @user.student
+    @student = User.where(role: 'Student').take
```

Comment:
It seems like the previous code was better as here we are getting a `student` only.